openstack/utils/values.yaml

```diff
@@ -6,6 +6,9 @@
 global:
   user_suffix: ""
   master_password: ""
+  # defined in values files so it can be updated automatically
+  # kubernetes_entrypoint_build_version:
+  #   4fbcf7ce324dc66e78480f73035e31434cfea1e8: latest
```

Comment:
If the version is practically non-human readable (a hash versus an actual version), I think we are better off using the content hash right away.

Reply:
I don't think I follow. Do you mean the content-hash of the snippet?
The "version" is defined in the shared-app-images pipeline as that's the "version" (whatever version means here) that's pulled from the upstream repository. We also have to manually bump it to comply with PCI staged deployments.

Reply:
Sorry, the correct term would be "image digest".

To expand: The proposal is to specify the image in the form:

`image: <repo>:<image-tag>` -> `image: repo:somefairlylonghash-dateandtimetothesecond`.

To my eyes, a tag in this format isn't really human-readable anymore, so we might as well go with:

`image: <repo>@<digest>` -> `image: repo@sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2`

I know, we do lose version pinning, but that container is mostly in maintenance mode anyway, so we might roll the whole thing forward as just one blob.

Reply:
Then we lose the automatic update on security-rebuild if we need to keep the hash here. The point of this proposal was to have explicit version updates by utils chart bumping while getting security updates (i.e. the last part of the tag updated) automatically.

If we don't want the automatic security updates of the base OS for the same upstream version rolled out automatically, I don't care how we track the image - if by tag or digest.

Reply:
I'll close this, as the topic seems to be more generic than how to store the tag / version.